python/lsst/summit/utils/plotting.py

```diff
@@ -0,0 +1,211 @@
+        case Image._maskedImage.MaskedImageF():
+            imData = inData.getImage().array
+        case Image._image.ImageF():
+            imData = inData.array
+        case Image._exposure.ExposureF():
+            match imageType:
+                case 'image':
+                    imData = inData.image.array
+                case 'masked':
+                    imData = inData.getMaskedImage().getImage().array
+        case _:
+            raise TypeError("This function accepts numpy array, lsst.afw.image.Exposure components. "
+                  f"Got {type(inData)}: {inData}")
+
+    # Add stretching
+    match stretch:
+        case 'css':
+            quantiles = utils.getQuantiles(imData, 256)
+            norm = colors.BoundaryNorm(quantiles, 256)
+        case 'asinh':
+            norm = vis.ImageNormalize(imData,
+                                      interval=vis.PercentileInterval(percentile),
+                                      stretch=vis.AsinhStretch(a=0.1))
+        case 'power':
+            norm = vis.ImageNormalize(imData,
+                                      interval=vis.PercentileInterval(percentile),
+                                      stretch=vis.PowerStretch(a=2))
+        case 'log':
+            norm = vis.ImageNormalize(imData,
+                                      interval=vis.PercentileInterval(percentile),
+                                      stretch=vis.LogStretch(a=1))
+        case 'linear':
+            norm = vis.ImageNormalize(imData,
+                                      interval=vis.PercentileInterval(percentile),
+                                      stretch=vis.LinearStretch())
+        case 'sqrt':
+            norm = vis.ImageNormalize(imData,
+                                      interval=vis.PercentileInterval(percentile),
+                                      stretch=vis.SqrtStretch())
+        case _:
+            norm = None
+
+    ax.imshow(imData, cmap=cmap, origin='lower', norm=norm)
+
+    # Add grid
+    if coordGrid:
+        ax.grid(color='white', ls='solid')
+
+    # Add compass
+    if compass:
+        color = 'r'
+        try:
+            wcs = inData.getWcs()
+        except AttributeError:
+            wcs = None
+
+        if wcs:
+            anchorRa, anchorDec = wcs.pixelToSky(anchorPix, anchorPix)
+            east = wcs.skyToPixel(geom.SpherePoint(anchorRa + 30.0 * geom.arcseconds, anchorDec))
+            north = wcs.skyToPixel(geom.SpherePoint(anchorRa, anchorDec + 30. * geom.arcseconds))
+
+            ax.arrow(anchorPix, anchorPix, north[0]-anchorPix, north[1]-anchorPix,
+                     head_width=1., head_length=1., color=color)
+            ax.arrow(anchorPix, anchorPix, east[0]-anchorPix, east[1]-anchorPix,
+                     head_width=1., head_length=1., color=color)
+            ax.text(north[0], north[1], 'N', color=color)
+            ax.text(east[0], east[1], 'E', color=color)
+
+    # Add centroids
+    if centroids:
```

Comment:
I think this whole approach to dealing with multiple types of sources to plot might not be the most advisable. I think it might be more simple to have a `footprints` arg which accepts either a list of footprints or a `footprintSet`, a `centroid` arg which takes a list of centroids, and a `sourceCat` arg which accepts sourceCatalogs. I think that would make it easier for users, and make this code a little less contrived and delicate.